lib/test_longest_symbol.c

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-2.0+
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/stringify.h>
+#include <linux/kprobe.h>
+#include <linux/kallsyms.h>
+#include "../tools/testing/selftests/kselftest/module.h"
+
+#define LONGEST_SYMBOL start_of_the_longest_symbol_possible__123456789_123456789_123456789_123456789_123456789_123__end_of_the_longest_symbol_possible
+
+KSTM_MODULE_GLOBALS();
+
+/*
+ * Kernel module for testing the longest kernel symbol
+ */
+
+void LONGEST_SYMBOL(void) {}
+EXPORT_SYMBOL(LONGEST_SYMBOL);
+
+_Static_assert((KSYM_NAME_LEN == strlen(__stringify(LONGEST_SYMBOL))), \
+		"LONGEST_SYMBOL not up to date with KSYM_NAME_LEN");
```

Comment:
I think the parenthesis can be removed:

```suggestion
_Static_assert(KSYM_NAME_LEN == strlen(__stringify(LONGEST_SYMBOL)), \
		"LONGEST_SYMBOL not up to date with KSYM_NAME_LEN");
```